### PR TITLE
Validate empty and duplicate OIDC audiences

### PR DESCRIPTION
**Pull request by mefarazath**
Fixes https://github.com/wso2/product-is/issues/11080

### CODE REVIEW

**Review by ayshsandu — APPROVED** [review]

**Review by IsuraD — APPROVED** [review]

**Review by ChamathNS — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 1]
```diff
@@ -331,6 +331,18 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
         return OAuthUtil.buildConsumerAppDTO(app);
     }
 
+    private String[] filterOIDCAudience(OAuthConsumerAppDTO application) {
+
+        if (application.getAudiences() != null) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:336]: On second thoughts, wouldn't it be better to fail the requests if the client sends invalid or duplicate audiences?

This behaviour will be consistent with CORS origin validation logic as well.

Comment (mefarazath) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:338]: Decide to fail the request if empty or duplicate audience values were sent

**Comment by mefarazath** [comment]
Do not merge this PR until the console has been fixed to not send empty or duplicate values

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 1 above)
Comment (ChamathNS) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:338]: +1 to fail the request if the client send invalid audiences.

**Review by ChamathNS — APPROVED** [review]

**Pull request merged**